### PR TITLE
added Bhashini API

### PR DESCRIPTION
**Pull request by luv-singh-ai**
added Bhashini API

### CODE REVIEW

File: notebooks/egov/pgr-create-curl.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+curl 'https://staging.digit.org/pgr-services/v2/request/_create?tenantId=pg.citya' \
+  -H 'authority: staging.digit.org' \
+  -H 'accept: application/json, text/plain, */*' \
+  -H 'accept-language: en-GB,en-US;q=0.9,en;q=0.8' \
```
Comment (gksoriginals) [inline notebooks/egov/pgr-create-curl.txt:4]: Please put only notebooks in this directory

File: notebooks/egov/complaint_openai_assistant.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
+import openai
+from openai._client import OpenAI
+client = OpenAI()
+
```
Comment (gksoriginals) [inline notebooks/egov/complaint_openai_assistant.py:4]: What is this file ?

File: notebooks/egov/Staging_PGR_Collection.json [hunk 3]
```diff
@@ -0,0 +1,231 @@
+{
+	"info": {
```
Comment (gksoriginals) [inline notebooks/egov/Staging_PGR_Collection.json:2]: Why do we need this ?

File: notebooks/egov/Staging_Environment_File.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+	"id": "f9572fec-6209-49cc-a8f4-a8846ea36d00",
+	"name": "Staging",
+	"values": [
```
Comment (gksoriginals) [inline notebooks/egov/Staging_Environment_File.json:4]: Why do we need this ?

File: main.py [hunk 5]
```diff
@@ -1,56 +1,249 @@
...
 async def respond_audio(update: Update, context: ContextTypes.DEFAULT_TYPE):
     audio_file = await context.bot.get_file(update.message.voice.file_id)
 
     # Use a temporary file
-    with tempfile.NamedTemporaryFile(suffix='.ogg', delete=True) as temp_audio_file:
+    with tempfile.NamedTemporaryFile(suffix='.wav', delete=True) as temp_audio_file:
         await audio_file.download_to_drive(custom_path=temp_audio_file.name)
         chat_id = update.effective_chat.id
-        print(chat_id)
-        response, history = audio_chat(chat_id, audio_file=open(temp_audio_file.name, "rb"))
```
Comment (gksoriginals) [inline main.py:43]: Please make sure existing functions are not edited. Add a new handler function for bhashini if needed.

File: .vscode/launch.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
```
Comment (gksoriginals) [inline .vscode/launch.json:3]: Add .vscode and .DS_store...etc to .gitignore file so that these files won't be uploaded to the repo.

**Review by gksoriginals — CHANGES_REQUESTED** [review]

**Comment by gksoriginals** [comment]
Please fix the conflicts as wel

File: notebooks/egov/pgr-create-curl.txt (hunk 1 above)
Comment (luv-singh-ai) [reply notebooks/egov/pgr-create-curl.txt:4]: done

File: notebooks/egov/complaint_openai_assistant.py (hunk 2 above)
Comment (luv-singh-ai) [reply notebooks/egov/complaint_openai_assistant.py:4]: these all were initial files egov people shared with us. I'm removed the entire directory now. There's no use for it for us now

File: notebooks/egov/Staging_Environment_File.json (hunk 4 above)
Comment (luv-singh-ai) [reply notebooks/egov/Staging_Environment_File.json:4]: same

File: main.py (hunk 5 above)
Comment (luv-singh-ai) [reply main.py:43]: yes

File: notebooks/egov/Staging_PGR_Collection.json (hunk 3 above)
Comment (luv-singh-ai) [reply notebooks/egov/Staging_PGR_Collection.json:2]: same

File: core/ai.py [hunk 7]
```diff
@@ -146,8 +134,10 @@ def chat(chat_id, input_message):
                         tool_outputs=tool_output_array
                     )
                     run, status = get_run_status(run, client, thread)
-
-                    message = get_assistant_message(client, thread.id)
+                    
+                    resp1 = get_assistant_message(client, thread.id)
+                    message = bhashini_output(resp1, lang=lang)
```
Comment (gksoriginals) [inline core/ai.py:139]: Please remove bhashini_output() from chat() function and call it in the bhashini_text_chat() function after calling chat(). In that way bhashini logic and llm chat module will stay separate. We can easily switch from Assistant API to langchain or maybe even open source models.

**Review by gksoriginals — CHANGES_REQUESTED** [review]

File: core/ai.py (hunk 7 above)
Comment (luv-singh-ai) [reply core/ai.py:267]: done

File: core/ai.py [hunk 8]
```diff
@@ -100,36 +111,141 @@ def chat(chat_id, input_message):
...
+                
+    return assistant_message, history
+
+def audio_chat(chat_id, audio_file):
+    input_message = transcribe_audio(audio_file, client)
+    print(f"The input message is : {input_message}")
+    assistant_message, history =  chat(chat_id, input_message)
+    return assistant_message, history
+
+def bhashini_chat(chat_id, input_message):
```
Comment (gksoriginals) [inline core/ai.py:179]: @luv-singh-ai why do we need bhashini_chat() function ? We can still use the existing chat() function inside bhashini_text_chat() right ? 

input text in some language -> bashini function for translating to english -> pass translated english message to chat() -> translate response from chat() to particular language 

this is possible right ?

**Review by gksoriginals — CHANGES_REQUESTED** [review]

**Pull request merged**